src/library/export/dlgtrackexport.h

```diff
@@ -0,0 +1,62 @@
+#ifndef DLGTRACKEXPORT_H
+#define DLGTRACKEXPORT_H
+
+#include <future>
+
+#include <QDialog>
+#include <QString>
+#include <QScopedPointer>
+#include <QTime>
+
+#include "configobject.h"
+#include "library/export/trackexport.h"
+#include "library/export/ui_dlgtrackexport.h"
+
+// A dialog for interacting with the track exporter in an interactive manner.
+// Handles errors and user interactions.
+class DlgTrackExport : public QDialog,
```

Comment:
This class does a bit much for a dialog. 
I would prefer to have a dialog class independent from the business logic that exports the files. 
the dialog class should only handle the user interactions.